### ISSUE TITLE
Add option to preserve alpha test coverge for alpha channel mipmaps

### ISSUE THREAD

**Issue by GoogleCodeExporter** [milkman2100/gimp-dds#25]
```
Alpha channel on mipmaps should be handled differently than color channels to 
preserve alpha test coverage: currently generated low-level mipmaps almost 
transparent if application use simple alpha test (texel_alpha < 
alpha_threshold).

Good description of this problem and available solutions presented here: 
http://the-witness.net/news/2010/09/computing-alpha-mipmaps/

```

Original issue reported on code.google.com by `rutsky.v...@gmail.com` on 13 May 2013 at 9:57

**Comment by GoogleCodeExporter** [comment]
```
I implemented described feature request, attaching patch against trunk version 
from Subversion.

Implementation similar to suggested in article [1], but it doesn't use 
subsampling. Patch tested with plain RGBA textures.

It performs well on my test textures. Also performance can be increased by 
using OpenMP.

[1] http://the-witness.net/news/2010/09/computing-alpha-mipmaps/

```

Original comment by `rutsky.v...@gmail.com` on 15 May 2013 at 3:56

Attachments:
- [preserve_alpha_test_coverage.patch](https://storage.googleapis.com/google-code-attachments/gimp-dds/issue-25/comment-1/preserve_alpha_test_coverage.patch)

**Comment by GoogleCodeExporter** [comment]
```
Sorry guys, haven't had much free time lately.  Thank you for the feature 
suggestion, and thank you Vladimir for the patch.  I will get it into trunk 
ASAP.
```

Original comment by `ski...@gmail.com` on 16 Jul 2013 at 8:14
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Your changes are included in the new 3.0.0 release.  Thank you so much for your 
contribution Vladimir!
```

Original comment by `ski...@gmail.com` on 28 Nov 2013 at 7:25
- Changed state: **Fixed**
